misc/experimental/rust/CMakeLists.txt

```diff
@@ -26,13 +26,12 @@ if (CELIX_RUST_EXPERIMENTAL)
     FetchContent_MakeAvailable(Corrosion)
```

Comment:
According to documentation, this method only work for CMake >= 3.19. 
Our current requirement is `cmake_minimum_required (VERSION 3.18)`.

Reply:
Good point, I add a cmake_minimum_required for the Rust PoC. I think in this case it is not needed to require on top level.

Reply:
I did a revert on this and merged with #634 instead for the updated `cmake_minimum_required` call